Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[main_jemalloc.go, line 24 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6239#-KFyuQ_HMvtJEPDftybK:-KFz2TM0mFqvICbbukQL:919482036) ([raw file](https://github.com/cockroachdb/cockroach/blob/02649eac42d7d90ec1c11ae1b968ae79ec70fc1c/main_jemalloc.go#L24)):*
ah. turns out I do need them. otherwise, I get:
```
# github.com/cockroachdb/cockroach
Undefined symbols for architecture x86_64:
  "_malloc_stats_print", referenced from:
      __cgo_238dea7fc55c_Cfunc_malloc_stats_print in main_jemalloc.cgo2.o
     (maybe you meant: __cgo_238dea7fc55c_Cfunc_malloc_stats_print)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6239)*
<!-- Sent from Reviewable.io -->

--- a/main_jemalloc.go
+++ b/main_jemalloc.go
@@ -1,0 +1,33 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+// +build jemalloc
+
+package main
+
+// #cgo darwin CPPFLAGS: -I../c-jemalloc/darwin_includes/internal/include
+// #cgo linux CPPFLAGS: -I../c-jemalloc/linux_includes/internal/include
+// #cgo darwin LDFLAGS: -Wl,-undefined -Wl,dynamic_lookup
+// #cgo linux LDFLAGS: -Wl,-unresolved-symbols=ignore-all
+//
+// #include <jemalloc/jemalloc.h>
+import "C"
+
+import _ "github.com/cockroachdb/c-jemalloc"
+
+func init() {
+	C.malloc_stats_print(nil, nil, nil)
+}

